src/v/cloud_storage/async_manifest_view.h

```diff
@@ -39,9 +39,34 @@
 
 namespace cloud_storage {
 
+struct async_view_timestamp_query {
+    async_view_timestamp_query(
+      kafka::offset min_offset_,
+      model::timestamp ts_,
+      kafka::offset max_offset_)
```

Comment:
nit: the underscores in arguments is a bit unconventional. Maybe just `min`, `t`, `max` or something?